app/grandchallenge/profiles/models.py

```diff
@@ -244,6 +245,41 @@ def dispatch_unread_direct_messages_email(
             subscription_type=EmailSubscriptionTypes.NOTIFICATION,
         )
 
+    @cached_property
+    def file_civs_user_has_permission_to_use(self):
+        # local import to avoid circular dependency
+        from grandchallenge.components.models import (
+            ComponentInterfaceValue,
+            InterfaceKind,
+        )
+
+        job_query = get_objects_for_user(
+            self.user, "algorithms.view_job"
+        ).filter(inputs__pk__in=OuterRef("pk"))
```

Comment:
They should also be able to use job outputs too right? I think evaluation outputs is also missing. This is quite similar to the logic in `serve_component_interface_value`, maybe they should be unified. We have to be extremely careful here as things like say, evaluation inputs, should not be downloadable or viewable by people with the `view_evaluation` permission. Maybe that permission should be changed but that is a separate issue.

Reply:
Then again, this is only about file CIVs, and the standard evaluation output is a metrics file that is stored in the DB (so a value CIV). So that wouldn't be included here anyway.

Reply:
> It would need to be tested with something roughly the size of production

So something like 2 million CIVs spready over algorithm inputs, outputs, display sets and archive items. I'll do some testing

Reply:
The users profile is for metadata about the user. This is about authorisation, which is a very different thing. It would be best to move this functionality to an independent function that takes the user. If the consumer is calling it many times it can cache it separately.

> A challenge admin can add outputs they don't want the participants to see.

We should be absolutely clear that we do not offer that functionality and we never have. We have row based permissions, there is no difference between any of the outputs, if someone can view the evaluation, they can view all of the outputs. This has always been the case. Now, you have the situation where someone could download the output, then re-upload it. We want to avoid that.

> Then again, this is only about file CIVs, and the standard evaluation output is a metrics file that is stored in the DB (so a value CIV).

We have many evaluations that output many things. The point here is about being consistent throughout the platform so that we do not have the same, but subtly different, business logic in different places in the codebase. We have to aim for consistency here, especially if we find bugs they need to be fixed in one place, and we shouldn't have to survey the entire codebase for these cases where we have multiple implementations of what should be the same thing. This is especially important for things as critical as authorisation as data security is our number 1 priority on the platform.

Reply:
> It would be best to move this functionality to an independent function that takes the user.

Ok, makes sense!

> We have row based permissions, there is no difference between any of the outputs, if someone can view the evaluation, they can view all of the outputs.

Yes, I know that. What I was thinking of when I wrote is the filtering of the metrics in metrics.json that you implemented a while ago, but this is not on the level of permissions of course. 

> Now, you have the situation where someone could download the output, then re-upload it. We want to avoid that.

I'm not sure I understand this point. So then evaluation outputs should not be downloadable, right? 

> The point here is about being consistent throughout the platform so that we do not have the same, but subtly different, business logic in different places in the codebase.

I completely agree. This needs to be unified in one place.

Reply:
> I'm not sure I understand this point. So then evaluation outputs should not be downloadable, right?

Evaluation outputs should be downloadable. If you did not include them in the 2nd permission implementation then people could download then upload, we want to avoid that pattern. This is why you're implementing this right?

Reply:
Ah, you're correct. Sorry! They were removed in fa5ae89f9e40ae2c0bbb83e48c9e072e2b7d5a68. Lets not change it.